### PR TITLE
Refactor wifi shell and samples

### PR DESCRIPTION
**Pull request by jukkar**
This PR relocates the wifi shell sample to `samples/net/wifi/shell`. This is done so that we can add more wifi samples to `samples/net/wifi` directory. One such sample can be found in `samples/boards/espressif/wifi_apsta_mode` directory. That sample is generic and can be used with other vendor wifi boards.

This is related to #80111

### CODE REVIEW

**Comment by jukkar** [comment]
I think we could try to get this to 4.0 because this just relocates samples around and does not introduce any new functionality.

**Review by rlubos — DISMISSED** [review]

**Review by sylvioalves — DISMISSED** [review]

**Review by krish2718 — DISMISSED** [review]

**Review by kartben — CHANGES_REQUESTED** [review]
Needs redirects.py entries for the moved READMEs
Needs a zephyr:code-sample-category in the new wifi folder (see samples/net/cloud)

**Comment by kartben** [comment]
+1 to try to get this in for 4.0, it's great to see the consolation of multiple wifi samples

File: samples/net/wifi/apsta_mode/README.rst [hunk 1]
```diff
@@ -1,13 +1,14 @@
-.. zephyr:code-sample:: esp32-wifi-ap-sta-mode
+.. zephyr:code-sample:: wifi-ap-sta-mode
    :name: Wi-Fi AP-STA mode
    :relevant-api: wifi_mgmt dhcpv4_server
 
-   Setup ESP32 to function as both an Access Point (AP) and a Station (STA) simultaneously.
+   Setup Wi-Fi supported board like ESP32 to function as both an Access Point (AP) and
+   a Station (STA) simultaneously.
```
Comment (kartben) [inline samples/net/wifi/apsta_mode/README.rst:6]: Can probably be made shorter:

```suggestion
   Configure a Wi-Fi board to operate as both an Access Point (AP) and a Station (STA).
```

**Comment by jukkar** [comment]
* updated according to comments

**Review by rlubos — DISMISSED** [review]

**Comment by kartben** [comment]
```
WARNING: samples/net/wifi/boards/cc3220sf_launchxl.conf not found in zephyr at '/home/runner/work/zephyr/zephyr/doc/_build/src/boards/ti/cc3220sf_launchxl/doc/index.rst:216'
WARNING: samples/net/wifi/boards/cc3235sf_launchxl.conf not found in zephyr at '/home/runner/work/zephyr/zephyr/doc/_build/src/boards/ti/cc3235sf_launchxl/doc/index.rst:216'
```

this is #80530 in action -- sorry that this means you need another push, but I am glad this is now helping catch this kind of dead references, tbh :)
(you also have the compliance issue re: sorted lines in redirects.py but I'm sure you saw that).

File: samples/net/wifi/README.rst [hunk 2]
```diff
@@ -1,48 +1,5 @@
...
-   4    | gksu                             4     | 1    | -26  | WPA/WPA2
-   ----------
-   Scan request done
-
-   shell> wifi connect "gksu" 4 SecretStuff
-   Connection requested
-   shell>
-   Connected
-   shell>
+   These samples demonstrate how to connect Zephyr devices to various Wi-Fi services.
```
Comment (kartben) [inline samples/net/wifi/README.rst:5]: just spotted this, sorry (and looks like you still need a push to fix the TI CC boards docs anyways)
wording doesn't sound right and looks like a hasty copy-paste from the Cloud (services) samples

```suggestion
   These samples demonstrate various Wi-Fi use cases for boards that support it.
```

or something along these lines?

**Review by kartben — APPROVED** [review]
https://builds.zephyrproject.io/zephyr/pr/80636/docs/samples/net/wifi/README.html

Looking good! :)
Thanks!

**Review by rlubos — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**